lab_4/main.py

```diff
@@ -2,46 +2,141 @@
 Lab 4
 """
 
+import re
 from ngrams.ngram_trie import NGramTrie
 
 
 def tokenize_by_sentence(text: str) -> tuple:
-    pass
+    if not isinstance(text, str):
+        raise ValueError
+
+    text = re.sub(r'[^A-Za-z\n.?! ]', '', text).lower()
+    text = re.sub(r'[!?.]', ' <END> ', text)
+
+    tokens = text.split()
+    if tokens and tokens[-1] != '<END>':
```

Comment:
Are you sure about `tokens[-1] != '<END>'`?

Reply:
It is written for the text == 'I am Ann'. (The sentence doesn't have the dot or smth else)